osgp/protocol-adapter-dlms/osgp-protocol-adapter-dlms/src/main/java/org/opensmartgridplatform/adapter/protocol/dlms/domain/commands/utils/FindMatchingChannelHelper.java

```diff
@@ -99,7 +99,7 @@ private static boolean validateMbusDeviceInfoMatch(final MbusChannelElementsDto
     private static boolean hasValidMbusAttributes(final MbusChannelElementsDto mbusChannelElements,
             final ChannelElementValuesDto channelElementValues) {
         return channelElementValues.isMbusSlaveDeviceConfigured()
-                && mbusChannelElements.getPrimaryAddress().equals(mbusChannelElements.getPrimaryAddress());
+                && mbusChannelElements.getPrimaryAddress().equals(channelElementValues.getPrimaryAddress());
```

Comment:
why did you change this?

Reply:
The equals method is here used to compare mbusChannelElements.getPrimaryAddress() to mbusChannelElements.getPrimaryAddress(), which is exactly the same object (comparing is redundant).

Looking at the other methods in this class, I had assumed that this was a copy/paste mistake and that the mbusChannelElements were supposed to be compared to the channelElementValues.